### PR TITLE
Clean up realm naming

### PR DESCRIPTION
**Pull request by peppy**
`RealmContextFactory` -> `RealmAccess`
`realmFactory` -> `realm`
`realmContextFactory` -> `realm`

Also most usage of `context` is replaced with `instance`.

### CODE REVIEW

**Comment by bdach** [comment]
Just going to note that I've resolved a conflict post-merge of #16548 in da77104. Resolution looks trivial but feel free to double check.

File: osu.Game.Benchmarks/BenchmarkRealmReads.cs [hunk 1]
```diff
@@ -27,9 +27,9 @@ public override void SetUp()
             storage = new TemporaryNativeStorage("realm-benchmark");
             storage.DeleteDirectory(string.Empty);
 
-            realmFactory = new RealmContextFactory(storage, "client");
+            realm = new RealmAccess(storage, "client");
 
-            realmFactory.Run(realm =>
+            realm.Run(realm =>
```
Comment (bdach) [inline osu.Game.Benchmarks/BenchmarkRealmReads.cs:32]: Some places in this pull look to have been changed to `realm.Run(r => ...)`, but this and a large number of other places (tens to hundreds left in this pull alone) continue (ab)using shadowing by writing `realm.Run(realm => ...)`. I presume this is just an oversight and another pass to fully eliminate the `realm.Run(realm => ...)` pattern is in order? Not a fan of name shadowing ever in general.

File: osu.Game/Tests/Visual/OsuTestScene.cs [hunk 2]
```diff
@@ -75,9 +75,9 @@ protected IAPIProvider API
         /// <remarks>
         /// In interactive runs (ie. VisualTests) this will use the user's database if <see cref="UseFreshStoragePerRun"/> is not set to <c>true</c>.
         /// </remarks>
-        protected RealmContextFactory ContextFactory => contextFactory.Value;
+        protected RealmAccess Access => contextFactory.Value;
```
Comment (bdach) [inline osu.Game/Tests/Visual/OsuTestScene.cs:78]: Not sure about this one, `Access` seems too non-specific. Would prefer `RealmAccess` (or maybe even `Realm`) if feasible.

File: osu.Game/Stores/RealmArchiveModelImporter.cs [hunk 3]
```diff
@@ -59,7 +59,7 @@ public abstract class RealmArchiveModelImporter<TModel> : IModelImporter<TModel>
 
         protected readonly RealmFileStore Files;
 
-        protected readonly RealmContextFactory ContextFactory;
+        protected readonly RealmAccess Access;
```
Comment (bdach) [inline osu.Game/Stores/RealmArchiveModelImporter.cs:62]: Similarly here, would swap out for `RealmAccess` or `Realm`.

File: osu.Game/Tests/Visual/OsuTestScene.cs (hunk 2 above)
Comment (peppy) [reply osu.Game/Tests/Visual/OsuTestScene.cs:78]: weird, that was rider alone.

File: osu.Game.Benchmarks/BenchmarkRealmReads.cs (hunk 1 above)
Comment (peppy) [reply osu.Game.Benchmarks/BenchmarkRealmReads.cs:32]: interesting that these don't trigger inspections..

File: osu.Game/Tests/Visual/OsuTestScene.cs (hunk 2 above)
Comment (peppy) [reply osu.Game/Tests/Visual/OsuTestScene.cs]: Should be resolved. I've gone with just `Realm` as it doesn't clash due to case.

File: osu.Game/Stores/RealmArchiveModelImporter.cs (hunk 3 above)
Comment (peppy) [reply osu.Game/Stores/RealmArchiveModelImporter.cs]: Also resolved as above.

File: osu.Game.Benchmarks/BenchmarkRealmReads.cs (hunk 1 above)
Comment (peppy) [reply osu.Game.Benchmarks/BenchmarkRealmReads.cs]: I've updated all remaining cases I could find. Still not 100% on the `r` variable, sometimes it feels a bit hard to follow, but let's go with it for now.

**Review by smoogipoo — APPROVED** [review]
